src/tools/compiletest/src/util.rs

```diff
@@ -14,43 +14,62 @@ use common::Config;
 /// Conversion table from triple OS name to Rust SYSNAME
 const OS_TABLE: &'static [(&'static str, &'static str)] = &[
     ("android", "android"),
+    ("androideabi", "android"),
     ("bitrig", "bitrig"),
     ("cloudabi", "cloudabi"),
     ("darwin", "macos"),
     ("dragonfly", "dragonfly"),
+    ("eabi", "eabi"),
+    ("eabihf", "eabi"),
+    ("elf", "elf"),
```

Comment:
Uh these seem rather odd. I know triples can be weird sometimes, are these actual "operating systems" or do they just happen to be in that position in the triple?

Reply:
Um, I guess a better question is: which triples do there come from?

Reply:
They did seem odd, but I didn't feel certain enough to exclude them. They come from`thumbv6m-none-eabi` (and the other `thumb*` triples) and `msp430-none-elf`. Do you think I should just exclude these?

Reply:
Ah so these triples have no "vendor" component, so the *second* component is the "OS" and the last component is the *ABI*.

Reply:
Ah — is it always safe to make this assumption? (I.e. elided component is vendor?)